spec/workers/sha_poller_worker_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe ShaPollerWorker, type: :worker do
+  before {
+    allow(AppEnv).to receive(:[]).with('HEROKU_SLUG_COMMIT') { nil }
+  }
+
+  describe "a poller" do
+    context "when deployed sha is different from a known sha" do
+      it "will publish the response when performing" do
```

Comment:
nit: test description is a little confusing.

Maybe something like this would be more accurate?:
 
SHA Poll worker
should publish the current SHA
when triggered